src/main/java/org/json/JSONArray.java

```diff
@@ -114,6 +138,9 @@ public JSONArray(JSONTokener x) throws JSONException {
                         throw x.syntaxError("Expected a ',' or ']'");
                     }
                     if (nextChar == ']') {
+                        if (jsonParserConfiguration.isStrictMode()) {
```

Comment:
Errors should indicate 'strict mode' in the text.
Comments should be added as approriate.